ui/src/data/models/Application/Display.ts

```diff
@@ -117,3 +120,42 @@ export class DisplayEdge implements DisplayEdgeShape, PathedData {
     );
   }
 }
+
+export class DisplayEconomicEvent {
```

Comment:
I'm confused about this. For one, there's already an implementation of `EconomicEvent` in `data/models/Valueflows/Observation.ts`:`EconomicEvent`.

The purpose of this file is act as a repository for the DisplayNode and DisplayEdge types that we store in a database. If there were any other objects that we store in the database and we display that have properties not in a Valueflows model, then we would put it here.

The `getNamesForId` seems like a helper that I've been putting in the `logic` directory. Continuing that tradition, the function should be renamed to `getAgentNamesForEconomicEvent`, rewritten to use the existing `EconomicEvent` model, and placed as a function in `ui/logic/flows.ts`.

I think that file is starting to get out of hand and we may need to start thinking about breaking it up into specific files used for specific views.